public-ui/src/app/data-browser/views/concept-chart/concept-chart-react.component.tsx

```diff
@@ -0,0 +1,407 @@
+                measurementGenderCountAnalysis = conceptAnalyses.measurementGenderCountAnalysis;
+                break;
+            default:
+                selectedAnalysis = conceptAnalyses.genderAnalysis;
+                break;
+        }
+
+        this.setState({
+            graphToShow: g,
+            selectedChartAnalysis: selectedAnalysis,
+            measurementGenderCountAnalysis: measurementGenderCountAnalysis,
+            displayGraphErrorMessage: selectedAnalysis === undefined
+        }, () => {
+             this.prepMeasurementChartData();
+         });
+    }
+
+    prepMeasurementChartData() {
+        const {graphToShow, measurementGenderCountAnalysis, conceptAnalyses} = this.state;
+        if (graphToShow === 'Values') {
+            const genderResults = conceptAnalyses.genderAnalysis.results;
+            const chartGenderOrder = ['8507', '8532', '0'];
+            genderResults.sort((a, b) => {
+                return chartGenderOrder.indexOf(a.stratum2) - chartGenderOrder.indexOf(b.stratum2);
+            });
+            const unitCounts = [];
+            for (const aa of measurementGenderCountAnalysis) {
+                      let sumCount = 0;
+                      for (const ar of aa.results) {
+                        sumCount = sumCount + ar.countValue;
+                      }
+                      unitCounts.push({ name: aa.unitName, count : sumCount});
+            }
+            unitCounts.sort((a, b) => {
+                return a.count - b.count;
+            });
+            let unitNames = unitCounts.map(d => d.name);
+            const noUnit = unitNames.filter(n => n.toLowerCase() === 'no unit');
+            unitNames = unitNames.filter(n => n.toLowerCase() !== 'no unit');
+            if (noUnit.length > 0) {
+                unitNames.push(noUnit[0]);
+            }
+            if (unitNames.length > 0) {
+                    this.setState({
+                        selectedUnit: unitNames[0],
+                        unitNames: unitNames,
+                        genderResults: genderResults
+                    }, () => {
+                        this.showMeasurementGenderHistogram(unitNames[0]);
+                    });
+            }
+        }
+        return;
+    }
+
+    showMeasurementGenderHistogram(unit: string) {
+        const {selectedChartAnalysis, measurementGenderCountAnalysis} = this.state;
+        let mixtureOfValues = false;
+        if (unit.toLowerCase() === 'no unit') {
+            const unitResults = selectedChartAnalysis.find(aa => aa.unitName === unit);
+            if (unitResults && unitResults.results && unitResults.results.length > 0) {
+                const numericResults = unitResults.results.filter(r => r.measurementValueType === 'numeric');
+                const textResults = unitResults.results.filter(r => r.measurementValueType === 'text');
+                if (numericResults && numericResults.length > 0 && textResults && textResults.length > 0) {
```

Comment:
These should at least be empty arrays since you checked `unitResults.results.length > 0` above, can simplify the conditions:
```suggestion
                if (numericResults.length > 0 && textResults.length > 0) {
```

Reply:
Changed this. thanks